### ISSUE TITLE
Determine non-breaking changes between revisions

### ISSUE THREAD

**Issue by dfaller** [ngageoint/scale#1267]
Implement the determination of whether a job/recipe revision change is breaking or not and have this determination used when doing a recipe diff. This would allow non-breaking job updates (change resources, etc) to not automatically mark nodes as "changed" and require reprocessing.

**Comment by emimaesmith** [comment]
We believe this means to give the operator more credit whether or not they want to re-run a parent of the changed job and only mark a recipe node as "changed" and in need of reprocessing if the interfaces are known to be incompatible (mismatched inputs/outputs, etc).

**Comment by JohnPTobe** [comment]
This is correct.  Since recipes are now pinned to a revision, when a sub-recipe or job type is updated we want to be smarter about whether a recipe needs to be reprocessed automatically.  If a job type interface change is just a small one and the output won't be changed it makes sense to mark the recipe as updated and notify the user if they want to re-process but don't automatically reprocess when the result won't change.  How to actually determine if a diff rises to the point of breaking is going to be tricky.
